### PR TITLE
Splatter Lands 1 Tile Away in Target Dir & Xeno-blood Splatter Colour Fix

### PR DESCRIPTION
**Pull request by KasparoVy**
### **Shamelessly regurgitating what I posted on the original feature PR and my PR on that feature's branch.**

Adjusts splattered blood to land 1 tile away from the mob in the direction of the graphic.
Splatter can land on walls, windows, etc. and works with diagonals.

Pictures:
bloodflung
respectwindows
bloodydudes

:cl:
tweak: Blood splattered by projectiles hitting a mob now land one tile away in the direction of the splatter graphic. Blood splattered on walls/windows/etc. can be cleaned off with soap.
fix: Xeno-blood splatter is now the appropriate colour.
/:cl:

### CODE REVIEW

**Comment by KasparoVy** [comment]
*shrug*

**Comment by Crazylemon64** [comment]
If it lands on a wall, does it become visible on both sides like wall fungus, or does it only show on one side like an APC?

**Comment by KasparoVy** [comment]
It becomes visible on both sides as it's actually sharing the wall's `loc` instead of being an adjacent `loc` on one side of the wall and pixel-shifted to look like it's on the wall.

**Comment by SamHPurp** [comment]
I think this is awesome, glad you PR'd it after the previous chat about it!

**Comment by KasparoVy** [comment]
Almost ready, just some final tweaks that'll have this system pixel shift splatters onto walls so you can only see them from the appropriate angles.

**Comment by KasparoVy** [comment]
respectwalls
@Crazylemon64 

Fixed.
I also fixed a bug where xeno blood splatter wasn't green.

Updating changelog.

**Comment by SamHPurp** [comment]
@KasparoVy so will Janiborgs running past the wall is going to automagically clean the wall due to the pixel shift?

Also, can you get bloody tracks by walking across the 'clean' square as well?

If not, this is now dramatically more complicated...

**Comment by KasparoVy** [comment]
@SamHPurp No to both, just tested it. I'm honestly quite surprised, but I won't question it.

**Comment by tigercat2000** [comment]
@KasparoVy Important question- Will this phase straight through glass? If you line someone up with a pane of glass and shoot them, are they going to get blood on the floor underneathe it that you have no chance of ever cleaning?

**Comment by scrubmcnoob** [comment]
*american stamp of approval*
also what tiger said
i hope blood splatters don't get stuck in funky places where they shouldn't be

File: code/game/objects/effects/overlays.dm [hunk 1]
```diff
@@ -205,6 +205,7 @@
 	animate(src, pixel_x = target_pixel_x, pixel_y = target_pixel_y, alpha = 0, time = duration)
 
 /obj/effect/overlay/temp/dir_setting/bloodsplatter/xenosplatter
+	color = null
```
Comment (Fox-McCloud) [inline code/game/objects/effects/overlays.dm]: Thanks. I'm a derp.

**Comment by SamHPurp** [comment]
> @SamHPurp No to both, just tested it. I'm honestly quite surprised, but I won't question it. 😆

Well thats a relief. Thumbs up for me... this even adds `slightly` more depth to Forensic play.

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -1882,3 +1882,27 @@ var/global/list/g_fancy_list_of_types = null
 		num_sample -= num
 		result += (1 << num)
 	return result
+
+/proc/pixel_shift_dir(var/dir, var/amount_x = 32, var/amount_y = 32) //Returns a list with pixel_shift values that will shift an object's icon one tile in the direction passed.
```
Comment (SamHPurp) [inline code/__HELPERS/unsorted.dm]: Useful helper, I might go back and update my Poster refactor to use this proc

**Comment by KasparoVy** [comment]
@tigercat2000 Good point, working on having it respect glass.

**Comment by KasparoVy** [comment]
@SamHPurp Seems that the blood on the wall not being affected was erroneous. It was indeed affected by borgs walking over/space cleaner reacting with the blood/tile the blood is 'projected' from.

Seems to work just fine now.
cleanedup
respectwindows

**Comment by SamHPurp** [comment]
Ah good work!

**Comment by Fox-McCloud** [comment]
conflicts.

Also, for blood on windows, how does that work for cleaning it off?

**Comment by KasparoVy** [comment]
If blood is on a wall/window you clean it by clicking it with soap. If it's visible, it can be soaped off.

I'll handle the conflicts in a couple hours

**Comment by KasparoVy** [comment]
Conflicts resolved.

**Comment by taukausanake** [comment]
Oh oh oh! Can you bloody people who are on the tile the blood will splatter to? That would be _amazing_.

Maybe another PR

**Comment by KasparoVy** [comment]
I'll keep that in mind. Maybe it can be something I'll tack on here, but I have 0 time.

**Comment by Tayyyyyyy** [comment]
This is just me nerding out about forensic science again but it'd be really cool if in the future the detective's scanner could pick up [info about an attack from the blood spatter](http://www.forensicsciencesimplified.org/blood/) on the ground.

**Comment by KasparoVy** [comment]
@Tayyyyyyy Done.
bloodydudes

**Comment by taukausanake** [comment]
DID THAT GREY JUST GET BLOOD SPLATTERED ON IT? :heart:

**Comment by KasparoVy** [comment]
Yep. I just took in I @'ed the wrong guy, too

My bad

**Comment by Aurorablade** [comment]
hey @KasparoVy not a fualt of your PR but can you see why some DNA transfer on blood procs may not be working. it does affect it in a way that if you hit two diffrent races and get the blood on the same spot (via gunshot) the color will maintain that of the first person shot. also some on mobs turn up white...pretty sure its some DNA list issue.

**Comment by Ar3nn** [comment]
to be fair, under windows wasn't magically uncleanable to begin with. Alt-clicking exists, blood icons did appear as objects in the list and you could then click it or the floor tile underneath to interact with it (clean it)

**Comment by tigercat2000** [comment]
@KasparoVy So, there's a bit of a bug with the blood system- specifically `bloody_body`. It doesn't always transfer the color of the blood correctly, for no apparent reason. As such, we'd appreciate it if you comment out that code until the blood system is fixed. Sorry for taking so long to get to this PR.

**Comment by KasparoVy** [comment]
@tigercat2000 Sorry for disappearing, I've not much time anymore and I totally forgot about this PR. I made the requested change, though I'm not sure if this was fixed at any point since then.

File: code/game/atoms.dm [hunk 3]
```diff
@@ -512,10 +512,11 @@ var/list/blood_splatter_icons = list()
 		overlays += blood_overlay
 
 /atom/proc/clean_blood()
-	germ_level = 0
-	if(islist(blood_DNA))
-		blood_DNA = null
-		return 1
+	if(!istype(src, /obj/effect/decal/cleanable/blood))
```
Comment (tigercat2000) [inline code/game/atoms.dm]: Heresy :eyes: (but I'll fix this myself because I don't want this to wait for Kasparov to fix again)

**Pull request merged**